src/app/frontend/common/pipes/safehtml.ts

```diff
@@ -15,17 +15,31 @@
 import {Pipe, SecurityContext} from '@angular/core';
 import {DomSanitizer, SafeHtml} from '@angular/platform-browser';
 
+const ansiColorClass = require('ansi-to-html');
+const ansiColor = new ansiColorClass();
+
 /**
  * Formats the given value as raw HTML to display to the user.
  */
 @Pipe({name: 'kdSafeHtml'})
 export class SafeHtmlFormatter {
   constructor(private readonly sanitizer: DomSanitizer) {}
 
-  transform(value: string): SafeHtml {
-    return this.sanitizer.sanitize(
+  transform(value: string, translate: string): SafeHtml {
```

Comment:
s/translate/mode

I'd also make it an enum.

```typescript
enum TextMode {
    Default,
    Colored
}
```

Then to make `mode` really optional assign it a default value of `TextMode.Default` and use `switch(mode)` to handle the logic.

Reply:
The enum and switch is now in place, thanks for taking a look at this so quickly.